frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -532,6 +549,30 @@ export class TransformResourceData {
     });
   };
 
+  getPipelineRunsForPipeline = (pipeline: Pipeline): PipelineRun[] => {
+    const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+    const pipelineRuns = _.get(this.resources, ['pipelineRuns', 'data']);
```

Comment:
This is the concern I have with blindly using `_.get`. It looks great on paper... but essentially you have just created a chance for a NPE. `_.get` allows you to blindly walk into an object without care, but you make no attempt to validate the response before using it.

Since this appears to be the structure of this utility class, I recommend just using the 3rd param here and setting it to an empty array so the `.filter` a couple lines down doesn't blow up if it fails to get something back (which likely will be `undefined`)

Reply:
@sahil143 I think we have had this conversation a couple times now... I'd like you to focus on your usage of `_.get` more closely. I worry we are going to slip and drop NPEs into our code with this kind of approach to `_.get`.

Either make liberal use of the 3rd param (making it the empty value of the type of what you're looking for) or avoid it at all costs, forcing the question of "what if the properties along the way are not there?".

Reply:
My bad, destructuring of object can be used here instead of _.get.
fixed in https://github.com/openshift/console/pull/3014/commits/3acef67b9193281dd53174625fd3d424051c286c

Reply:
Boy, not a lot of types going around this utility, huh?

Do we need to update the `PodRingResources` interface-type?
```
export interface PodRingResources {
  pods: Resource;
  replicaSets: Resource;
  replicationControllers: Resource;
  deployments?: Resource;
  deploymentConfigs?: Resource;
}
```

Reply:
fixed in https://github.com/openshift/console/pull/3014/commits/657b83830153a7aaf5401872fe3075a6ef553c11

Reply:
No, we don't need to update the PodRingResource types here because that only deals with getting the data for pods. 
> Is it possible for pipelineRuns not to be part of the this.resources object?
No, We added pipeline and Pipeline Run here https://github.com/openshift/console/pull/3014/files#diff-0dffb9abd12f8591b1d227efedf17dcbR113 in the resource object.

Reply:
Right you are ... my IDE had a reference issue where it didn't track it to this import lol. `@console` references seem to be an issue for my IDE to track.